backend/stylists/app/Http/Mapper/TipMapper.php

```diff
@@ -0,0 +1,250 @@
+            $tip->$dropdown_field = isset($request->$dropdown_field) && $request->$dropdown_field != '' ? $request->$dropdown_field : '';;
+        }
+        return $tip;
+    }
+
+    public function saveEntities($tip_id, $products, $looks)
+    {
+        $entity_product_ids = $products ? explode(',', $products) : [];
+        $entity_look_ids = $looks ? explode(',', $looks) : [];
+
+        $tip_entities = $this->getExistingEntities($tip_id);
+
+        $existing_look_entities = [];
+        $existing_product_entities = [];
+
+        $entity_type_look = EntityType::LOOK;
+        $entity_type_product = EntityType::PRODUCT;
+
+        if (count($tip_entities) > 0) {
+            foreach ($tip_entities as $tip_entity){
+                if ($tip_entity->entity_type_id == $entity_type_look) {
+                    array_push($existing_look_entities, $tip_entity->entity_id);
+                }elseif ($tip_entity->entity_type_id == $entity_type_product) {
+                    array_push($existing_product_entities, $tip_entity->entity_id);
+                }
+            }
+        }
+
+        $products_to_delete = array_diff($existing_product_entities, $entity_product_ids);
+        $looks_to_delete = array_diff($existing_look_entities, $entity_look_ids);
+        $products_to_add = array_diff($entity_product_ids, $existing_product_entities);
+        $looks_to_add = array_diff($entity_look_ids, $existing_look_entities);
+
+        $entity_array = [];
+        $index = 0;
+        foreach ($products_to_add as $entity_product_id) {
+            $entity_array[$index++] = array(
+                'tip_id' => $tip_id,
+                'entity_type_id' => $entity_type_product,
+                'entity_id' => $entity_product_id,
+            );
+        }
+
+        foreach ($looks_to_add as $entity_look_id) {
+            $entity_array[$index++] = array(
+                'tip_id' => $tip_id,
+                'entity_type_id' => $entity_type_look,
+                'entity_id' => $entity_look_id,
+            );
+        }
+
+        $status = true;
+        $message = '';
+        if (count($entity_array) > 0) {
+            try {
+                TipEntity::insert($entity_array);
+            } catch (\Exception $e) {
+                $status = false;
+                $message = $e->getMessage();
+                return array(
+                    'status' => $status,
+                    'message' => $message,
+                );
+            }
+        }
+
+        $raw_query = '';
+        foreach ($products_to_delete as $item) {
+            $raw_query = $raw_query . " OR (tip_id = '{$tip_id}' AND entity_type_id = '{$entity_type_product}' AND entity_id = '{$item}')";
+        }
+        foreach ($looks_to_delete as $item) {
+            $raw_query = $raw_query . " OR (tip_id = '{$tip_id}' AND entity_type_id = '{$entity_type_look}' AND entity_id = '{$item}')";
+        }
+
+        if (!empty($raw_query)) {
```

Comment:
Keep both insert and delete statements together in a single try catch. Also use a transaction to execute these queries.